services/job_runner.go

```diff
@@ -138,9 +138,7 @@ func startTask(
 	adapter, err := adapters.For(tr.Task, store)
 
 	if err != nil {
-		tr.Status = models.RunStatusErrored
-		tr.Result.SetError(err)
-		return tr
+		return tr.ApplyResult(tr.Result.SetError(err))
```

Comment:
Isn't `RunResult#SetError` now the same as `RunResult#WithError`? I think we should remove `SetError` altogether because it definitely misleads one into thinking it's mutating the caller.

Reply:
Great call. Updated in last commit